pydicom/tests/test_dataset.py

```diff
@@ -1135,10 +1135,10 @@ def test_save_as_compressed_no_encaps(self):
         ds.PixelData = b'\x00\x01\x02\x03\x04\x05\x06'
         ds['PixelData'].VR = 'OB'
         msg = (
-            r"The dataset uses a compressed Transfer Syntax UID but the "
-            r"Pixel Data hasn't been encapsulated"
+            r"Pixel Data has an undefined length indicating "
+            r"that it's compressed but the data isn't encapsulated"
         )
-        with pytest.warns(UserWarning, match=msg):
+        with pytest.raises(ValueError, match=msg):
```

Comment:
You have to remind me - does undefined length guarantee compression?  I thought the converse was true, but is it always the case in this direction?

Reply:
As far as I know, this is true. Uncompressed pixel data never have sequence delimiter items (which they would need in case of undefined length).

Reply:
PS3.5 A.4
> Pixel Data (7FE0,0010) may be encapsulated or native.

> If native, it shall have a defined Value Length

> If encapsulated (...)

> The Length of the Data Element (7FE0,0010) shall be set to the Value for Undefined Length (FFFFFFFFH)

Only compressed transfer syntaxes may be encapsulated (A.4.1-X).

Sequences may also be of undefined length, though

Reply:
> If native, it shall have a defined Value Length

This was what I was looking for, but for some reason couldn't find - thanks!